runtime/bcutil/cfreader.c

```diff
@@ -2167,11 +2167,15 @@ checkClass(J9PortLibrary *portLib, J9CfrClassFile* classfile, U_8* segment, U_32
 	U_32 i;
 	I_32 maxBootstrapMethodIndex = -1;
 
-	if(checkPool(classfile, segment, (U_8*)10, &maxBootstrapMethodIndex, flags)) {
+	if(checkPool(classfile, segment, (U_8*)10, &maxBootstrapMethodIndex, vmVersionShifted, flags)) {
 		return -1;
 	}
 
-	value = classfile->accessFlags & CFR_CLASS_ACCESS_MASK;
+	if (vmVersionShifted >= BCT_Java9MajorVersionShifted) {
```

Comment:
@DanHeidinga is this what you meant for the runtime check?

Reply:
yes